rules/select_file.bzl

```diff
@@ -0,0 +1,33 @@
+"""
+select_file() build rule implementation.
+
+Selects a single file from the outputs of some target by given relative path.
+"""
+
+def _impl(ctx):
+    if ctx.attr.subpath and len(ctx.attr.subpath) == 0:
+        fail("Subpath can not be empty.")
+
+    out = None
+    canonical = ctx.attr.subpath.replace("\\", "/")
+    for file_ in ctx.attr.srcs.files.to_list():
+        if file_.path.replace("\\", "/").endswith(canonical):
+            out = file_
+            break
+    if not out:
+        fail("Can not find specified file in '%s'" % str(ctx.attr.srcs))
```

Comment:
this would print out a wall of text of `str(ctx.attr.srcs)` is a big list, right? perhaps print it with `"\n".join`?

Reply:
Done some prettifying work:)